### PR TITLE
update eventDataIB and add eventReport, eventReports

### PR DESCRIPTION
**Pull request by yunhanw-google**
#### Problem
Update EventDataIB/EventReportIB/EventReports  encoding per latest IM spec and update the corresponding EventManagement.

#### Change overview
See above

#### Testing
Add unit tests for EventDataIB/EventReportIB/EventReports

### CODE REVIEW

File: src/app/EventManagement.cpp [hunk 1]
```diff
@@ -312,9 +315,32 @@ CHIP_ERROR EventManagement::ConstructEvent(EventLoadOutContext * apContext, Even
     VerifyOrExit(apOptions != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(apOptions->mTimestamp.mType != Timestamp::Type::kInvalid, err = CHIP_ERROR_INVALID_ARGUMENT);
 
-    eventDataElementBuilder.Init(&(apContext->mWriter));
-    eventPathBuilder = eventDataElementBuilder.CreateEventPath();
-    err              = eventPathBuilder.GetError();
+    eventReportBuilder.Init(&(apContext->mWriter));
+    // TODO: Update IsUrgent
+    // TODO: Update statusIB
```
Comment (turon) [inline src/app/EventManagement.cpp:320]: Could you generate github issues for these and track with `TODO(#xxxx)`?

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/EventManagement.cpp (hunk 1 above)
Comment (yunhanw-google) [reply src/app/EventManagement.cpp]: Done, thanks, updated
https://github.com/project-chip/connectedhomeip/issues/11386
https://github.com/project-chip/connectedhomeip/issues/11388

File: src/app/EventManagement.cpp [hunk 2]
```diff
@@ -583,31 +613,46 @@ CHIP_ERROR EventManagement::CopyEvent(const TLVReader & aReader, TLVWriter & aWr
 {
     TLVReader reader;
     TLVType containerType;
+    TLVType containerType1;
+    TLVType containerType2;
```
Comment (yufengwangca) [inline src/app/EventManagement.cpp:617]: It looks the containerType1 and containerType2 are used one after another, could we just reuse?

File: src/app/EventManagement.cpp [hunk 3]
```diff
@@ -583,31 +613,46 @@ CHIP_ERROR EventManagement::CopyEvent(const TLVReader & aReader, TLVWriter & aWr
...
-    err = aWriter.EndContainer(containerType);
-    SuccessOrExit(err);
-
-    err = aWriter.Finalize();
-    SuccessOrExit(err);
-
-exit:
-    return err;
+    ReturnErrorOnFailure(err);
+    ReturnErrorOnFailure(aWriter.EndContainer(containerType2));
```
Comment (yufengwangca) [inline src/app/EventManagement.cpp:652]: where aWriter StartContainer for containerType2

**Review by yufengwangca — APPROVED** [review]

File: src/app/EventManagement.cpp (hunk 3 above)
Comment (yunhanw-google) [reply src/app/EventManagement.cpp:652]: aWriter.StartContainer(TLV::ContextTag(to_underlying(EventReportIB::Tag::kEventData)), kTLVType_Structure, containerType2)

File: src/app/EventManagement.cpp (hunk 2 above)
Comment (yunhanw-google) [reply src/app/EventManagement.cpp:617]: yep, done, thanks

**Review by LuDuda — APPROVED** [review]

**Pull request merged**